addons/im_livechat/static/src/components/discuss/discuss_tests.js

```diff
@@ -28,33 +21,22 @@ QUnit.module('discuss_tests.js', {
         };
     },
     afterEach() {
-        utilsAfterEach(this);
-        if (this.widget) {
-            this.widget.destroy();
-        }
+        afterEach(this);
     },
 });
 
-QUnit.test('livechat in the sidebar', async function (assert) {
+QUnit.test('livechat in the sidebar: basic rendering', async function (assert) {
     assert.expect(5);
 
-    this.data.initMessaging = {
-        channel_slots: {
-            channel_livechat: [{
-                channel_type: "livechat",
-                id: 1,
-                is_pinned: true,
-                livechat_visitor: {
-                    country: false,
-                    id: false,
-                    name: "Visitor",
-                },
-            }],
+    // channel that is expected to be found in the sidebar
+    this.data['mail.channel'].records.push({
+        channel_type: 'livechat', // channel is expected to be livechat
+        id: 11, // random unique id, will be referenced in the test
+        livechat_visitor: {
+            name: "Visitor 11", // random name, will be asserted during the test
```

Comment:
The goal of the mock server is to simulate the server, and if we ever get to the point of actually using the server, what we will do is to call `create` which is the equivalent of `push` here. We will never flush a whole table before doing `create`.

This is even more important for some models where existing data must be there for the whole thing to work (partners, users, message types, activity types, ...).

So to avoid any confusion, it is much safer to always do `push`. If there are ever case where existing data are problematic, that means they shouldn't have been set as existing data in the first place.

Reply:
Ok to use `.push()`.